### PR TITLE
Move to using GraphQL mutations for SdmGoal and SdmGoalSet

### PR DESCRIPTION
**Pull request by cdupuis**
This relates to #788

### CODE REVIEW

File: lib/api-helper/goal/storeGoals.ts [hunk 1]
```diff
@@ -29,26 +29,28 @@ import {
...
     SdmGoalState,
-    SdmGoalWithPushFields,
+    UpdateSdmGoalMutation,
+    UpdateSdmGoalMutationVariables,
+    UpdateSdmGoalSetMutation,
+    UpdateSdmGoalSetMutationVariables,
 } from "../../typings/types";
 
+// tslint:disable-next-line:no-var-requires
+const omitEmpty = require("omit-empty");
```
Comment (ddgenome) [inline lib/api-helper/goal/storeGoals.ts]: Why not `import`? It seems the types are available.

File: lib/api-helper/listener/goalSetListener.ts [hunk 2]
```diff
@@ -44,11 +45,12 @@ export const GoalSetGoalCompletionListener: GoalCompletionListener = async gcl =
     if (result && result.SdmGoalSet && result.SdmGoalSet.length === 1) {
         const goalSet = result.SdmGoalSet[0];
         if (goalSet.state !== state) {
-            const newGoalSet = {
+            const newGoalSet: any = {
```
Comment (ddgenome) [inline lib/api-helper/listener/goalSetListener.ts]: Is it not a goal set?

File: lib/api-helper/goal/storeGoals.ts (hunk 1 above)
Comment (cdupuis) [reply lib/api-helper/goal/storeGoals.ts]: For whatever reason I couldn’t get the types to work. I can try again but I think they are broken. I’ll try again.

Comment (ddgenome) [reply lib/api-helper/goal/storeGoals.ts]: Like I said, not a big deal. If you are not using them, it might be worth uninstalling them.

**Pull request merged**